spec/mongoid/query_cache_spec.rb

```diff
@@ -123,6 +123,27 @@
         end
       end
     end
+
+    context "when cache is enabled and second batch of items is requested" do
+
+      before do
+        Mongoid::QueryCache.enabled = true
```

Comment:
The problem here is that setting ``Mongoid::QueryCache`` to enabled is a global setting, regardless of where you change it. So yes, in this case, the changes in the ``before`` block will persist after the context scope is left.

It's up to you whether you use around or before and after.